src/silverlabnwb/nwb_file.py

```diff
@@ -583,15 +586,37 @@ def add_stimulus(self):
             self.nwb_file.add_stimulus(TimeSeries(**attrs))
         self._write()
 
-    def read_cycle_relative_times(self, file_path):
+    def read_cycle_relative_times(self, folder_path):
         """Read the 'Single cycle relative times.txt' file and store the values in memory.
 
         Note that while the file has times in microseconds, we convert to seconds for consistency
         with other timestamps in NWB.
         """
-        assert os.path.isfile(file_path)
-        self.cycle_relative_times = pd.read_csv(file_path, names=('RelativeTime', 'CycleTime'),
-                                                sep='\t', dtype=np.float64) / 1e6
+
+        def rel(file_name):
+            """Return the path of a file name relative to the Labview folder."""
+            return os.path.join(folder_path, file_name)
+
+        roi_path = rel('ROI.dat')
+        assert os.path.isfile(roi_path)
+        if self.labview_version is LabViewVersions.pre2018:
+            file_path = rel('Single cycle relative times.txt')
+            assert os.path.isfile(file_path)
+            timings = LabViewTimingsPre2018(file_path, roi_path, self.imaging_info.dwell_time/1e6)
+            self.raw_pixel_time_offsets = timings.pixel_time_offsets
+            self.cycle_time = timings.cycle_time
+        elif self.labview_version is LabViewVersions.v231:
+            file_path = rel('Single cycle relative times_HW.txt')
+            assert os.path.isfile(file_path)
+            timings = LabViewTimings231(file_path,
+                                        roi_path=roi_path,
+                                        n_cycles_per_trial=self.imaging_info.cycles_per_trial,
+                                        n_trials=len(self.trial_times),
+                                        dwell_time=self.imaging_info.dwell_time/1e6)
+            self.raw_pixel_time_offsets = timings.pixel_time_offsets
+            self.cycle_time = timings.cycle_time
+        else:
+            raise ValueError('Unsupported LabView version for timings {}.'.format(self.labview_version))
```

Comment:
This is fine for now, but if you want to avoid it, we can add a method to the (abstract) `Timings` class which creates an object of the appropriate type based on the version we pass it (similar to how the abstract `Header` class works, if I remember right). If we pass the wrong version (or the wrong arguments for that version), we can throw the error there.

This can wait until later, however.